### PR TITLE
[CDAP-4573] MapReduce archetype and documentation

### PR DESCRIPTION
**Pull request by anew**
Jira: https://issues.cask.co/browse/CDAP-4573
This is to help people set up a MapReduce app with the correct dependencies and excludes.

### CODE REVIEW

File: cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+          <artifactId>netty</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>${hadoop.version}</version>
+      <scope>provided</scope>
+    </dependency>  </dependencies>
```
Comment (chtyim) [inline cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml]: Missing new line.

File: cdap-archetypes/cdap-mapreduce-archetype/README.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+Creating
+========
+
+To create a project from the archetype, use this script as an example
+(substituting for ${project.version} and ${your.hadoop.version} as appropriate)::
+
+  mvn archetype:generate 					
+    -DarchetypeGroupId=co.cask.cdap 			
+    -DarchetypeArtifactId=cdap-mapreduce-archetype
+    -DarchetypeVersion=${project.version}
```
Comment (chtyim) [inline cdap-archetypes/cdap-mapreduce-archetype/README.rst]: The archetypeVersion should be the version of the cdap archetype, which is 3.3.0-SNAPSHOT in this case and become 3.3.0 when released. The version that user provides for his own project is under `-Dversion`

File: cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    <app.main.class>${package}.MapReduceApp</app.main.class>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <cdap.version>3.3.0-SNAPSHOT</cdap.version>
+    <slf4j.version>1.7.5</slf4j.version>
+    <guava.version>13.0.1</guava.version>
+    <junit.version>4.11</junit.version>
+    <hadoop.version>${hadoopVersion}</hadoop.version>
+  </properties>
+
+  <repositories>
```
Comment (chtyim) [inline cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml]: For archetype generated project, we shouldn't includes repository as everything should be from maven central repo.

**Comment by chtyim** [comment]
Tried out the archetype, it works. There is one minor issue that "mvn package" failed due to unit-test failure. I think it's because in the unit-test we start the program and stop it immediately (as part of `@After` method from the `TestBase`) and there is a race that the stop cannot find the program from the runtime info to stop.

File: cdap-archetypes/cdap-mapreduce-archetype/README.rst (hunk 2 above)
Comment (anew) [inline cdap-archetypes/cdap-mapreduce-archetype/README.rst]: fixed. Actually this was copied from the cdap-app-archetype and I am fixing it there, too.

File: cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml (hunk 1 above)
Comment (anew) [inline cdap-archetypes/cdap-mapreduce-archetype/src/main/resources/archetype-resources/pom.xml]: fixed

**Comment by anew** [comment]
fixed the unit test issue by commenting the start of the MapReduce.

**Comment by anew** [comment]
@chtyim I addressed your comments. Please have another look

**Comment by chtyim** [comment]
LGTM

**Comment by awholegunch** [comment]
I found one typo, and fixed an old formatting error in the use of em-dahes. Please check. LGTM. 
I have kicked off a Quick Build: http://builds.cask.co/browse/CDAP-DOB83-1 please check in a bit if everything is OK...

**Comment by awholegunch** [comment]
Quick Build failed, but the error is in the cdap-default.xml, which you are not changing. So it LGTM.

**Comment by anew** [comment]
@awholegunch your change looks ok to me.

**Comment by anew** [comment]
squashed everything into a single commit and merged.

**Pull request merged**